lms/djangoapps/grades/tasks.py

```diff
@@ -61,13 +62,41 @@ class _BaseTask(PersistOnFailureTask, LoggedTask):  # pylint: disable=abstract-m
 
 
 @task(base=_BaseTask)
-def compute_grades_for_course(course_key, offset, batch_size):
+def compute_grades_for_course_with_kwargs(course_key, offset, batch_size, **kwargs):
+    """
+    TODO: Roll this back into compute_grades_for_course once all workers have
+    the version with **kwargs.
+
+    Wrapper for compute_grades_for_course.  Sets the ESTIMATE_FIRST_ATTEMPTED
+    flag, then calls the original task as a synchronous function.
+
+    This lets us add the estimate_first_attempted keyword argument without
+    worrying about forward compatibility of existing task workers.
+
+    estimate_first_attempted:
+        controls whether to override the value of the ESTIMATE_FIRST_ATTEMPTED
+        waffle switch.  If None, don't override, otherwise, set to the passed
+        boolean value.
+    """
+    estimate_first_attempted = kwargs.pop('estimate_first_attempted', None)
+    if isinstance(estimate_first_attempted, bool):
```

Comment:
Please create a sandbox and make sure the waffle flag is honored across processes.   Or, `run_all_servers` on devstack.